cmake/sdlplatform.cmake

```diff
@@ -6,17 +6,18 @@ macro(SDL_DetectCMakePlatform)
   elseif(PSP)
     set(SDL_CMAKE_PLATFORM psp)
   elseif(APPLE)
-    if(CMAKE_SYSTEM_NAME MATCHES ".*Darwin.*")
-      set(SDL_CMAKE_PLATFORM Darwin)
-    elseif(CMAKE_SYSTEM_NAME MATCHES ".*MacOS.*")
-      set(SDL_CMAKE_PLATFORM MacosX)
+    if(CMAKE_SYSTEM_NAME MATCHES ".*(Darwin|MacOS).*")
+      set(SDL_CMAKE_PLATFORM macOS)
     elseif(CMAKE_SYSTEM_NAME MATCHES ".*tvOS.*")
       set(SDL_CMAKE_PLATFORM tvOS)
     elseif(CMAKE_SYSTEM_NAME MATCHES ".*iOS.*")
+      set(SDL_CMAKE_PLATFORM watchOS)
+    elseif(CMAKE_SYSTEM_NAME MATCHES ".*watchOS.*")
       set(SDL_CMAKE_PLATFORM iOS)
     elseif (CMAKE_SYSTEM_NAME MATCHES "visionOS")
       set(SDL_CMAKE_PLATFORM visionOS)
-      set(VISIONOS ON)       # CMAKE does not set this automatically yet
```

Comment:
Does minimum cmake version we require set this automatically now?

Reply:
No, at the bottom of this function, our cmake script uppercases the value of `SDL_CMAKE_PLATFORM`, and set the variable to a true-ish value.

Reply:
OK